core/src/com/unciv/logic/battle/Battle.kt

```diff
@@ -226,21 +222,76 @@ object Battle {
+            ) return false
+
+            val captureChance = min(0.8f, 0.1f + attacker.getAttackingStrength().toFloat() / defender.getDefendingStrength().toFloat() * 0.4f)
+            return Random().nextFloat() <= captureChance
+        }
+
+        fun unitCapturedGermanyUnique():Boolean{
+            if (!defender.getCivInfo().isBarbarian() || attackedTile.improvement != Constants.barbarianEncampment)
+                return false
+
+            var unitCaptured = false
+            // German unique - needs to be checked before we try to move to the enemy tile, since the encampment disappears after we move in
+
+            for (unique in attacker.getCivInfo().getMatchingUniques(UniqueType.GainFromEncampment)) {
+                attacker.getCivInfo().addGold(unique.params[0].toInt())
+                unitCaptured = true
+            }
+            return unitCaptured
+        }
+
+
+        fun unitCapturedOttomanUnique():Boolean{
+            var unitCaptured = false
+            for (unique in attacker.getCivInfo().getMatchingUniques(UniqueType.GainFromDefeatingUnit)) {
+                if (defender.unit.matchesFilter(unique.params[0])
+                        && attacker.isMelee()
+                ) {
+                    attacker.getCivInfo().addGold(unique.params[1].toInt())
+                    unitCaptured = true
+                }
+            }
+            return unitCaptured
+        }
 
-        if (attacker.unit.getMatchingUniques(UniqueType.KillUnitCapture)
-            .none { defender.matchesCategory(it.params[0]) }
-        ) return false
+        val wasUnitCaptured = unitCapturedPrizeShipsUnique() || unitCapturedGermanyUnique() || unitCapturedOttomanUnique()
 
-        val captureChance = min(0.8f, 0.1f + attacker.getAttackingStrength().toFloat() / defender.getDefendingStrength().toFloat() * 0.4f)
-        if (Random().nextFloat() > captureChance) return false
+        if (!wasUnitCaptured) return false
 
         // This is called after takeDamage and so the defeated defender is already destroyed and
         // thus removed from the tile - but MapUnit.destroy() will not clear the unit's currentTile.
         // Therefore placeUnitNearTile _will_ place the new unit exactly where the defender was
-        return spawnCapturedUnit(defender.getName(), attacker, defender.getTile(), "Your [${attacker.getName()}] captured an enemy [${defender.getName()}]!")
+        return spawnCapturedUnit(defender.getName(), attacker, defender.getTile())
+    }
+
+    /** Places a [unitName] unit near [tile] after being attacked by [attacker].
+     * Adds a notification to [attacker]'s civInfo and returns whether the captured unit could be placed */
+    private fun spawnCapturedUnit(unitName: String, attacker: ICombatant, tile: TileInfo): Boolean {
+        val addedUnit = attacker.getCivInfo().placeUnitNearTile(tile.position, unitName) ?: return false
+        addedUnit.currentMovement = 0f
+        addedUnit.health = 50
```

Comment:
`addedUnit.health` as a function argument could solve #7376 in the same PR :)

Reply:
That needs to be dealt with separately anyway, since apparently it doesn't actually capture a unit but instead activates when capturing an encampment even if there's no unit... so that'll need to be another PR anyway